draft-ietf-mls-protocol.md

```diff
@@ -823,11 +825,14 @@ values:
 | pk(ns\[1\])  | E(pk(C), ps\[1\]), E(pk(D), ps\[1\]) |
 | pk(ns\[0\])  | E(pk(A), ps\[0\])                    |
 
-In this table, the value pk(X) represents the public key
-derived from the node secret X.  The value E(K, S) represents
+In this table, the value pk(ns\[X\]) represents the public key
+derived from the node secret X, whereas pk(X) represents the public leaf key
+for user X.  The value E(K, S) represents
 the public-key encryption of the path secret S to the
 public key K.
 
+After applying the commit, all group members MUST delete outdated path secrets.
```

Comment:
Unclear whether "outdated" here means "from before the Commit" or "transmitted in the Commit, but now obsolete because the keys have been generated from them".  Like I said above, I think we can focus on the latter (so that you'll never even have the former stored), we just need to make sure that the wording of the requirement doesn't interfere with the secondary uses of the path secrets.

Reply:
Do you mind suggesting phrasing here? This doesn't match my mental model of "when replacing key material in the tree after receiving a `DirectPath`, the outdated key material (that was replaced by key material in the `DirectPath` message) should be deleted, but there could be something I'm profoundly missing.

Reply:
The question is: if your nodes have `(secret, sk, pk)`, and you move from `(secret0, sk0, pk0) -> (secret1, sk1, pk1)`, do you need to delete secret0, secret1, or both?  (If you delete `secret1`, then you never have a `secret0` to delete)